src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs

```diff
@@ -352,5 +352,41 @@ public void It_rewrites_the_apphost_for_non_single_file_publish()
 
             appHostSize.Should().BeLessThan(singleFileSize);
         }
+
+        [CoreMSBuildOnlyTheory]
+        [InlineData("netcoreapp3.0", false)]
+        [InlineData("netcoreapp3.0", true)]
+        [InlineData("netcoreapp3.1", false)]
+        [InlineData("netcoreapp3.1", true)]
+        [InlineData("netcoreapp5.0", false)]
+        [InlineData("netcoreapp5.0", true)]
+        public void It_runs_single_file_apps(string targetFramework, bool selfContained)
```

Comment:
Could you change the assertion(assert different files are present or not) to reflect:

_In .net 5, single-file apps will leave native binaries unbundled in the publish directory by default.
Since Single-file bundles are processed in the framework, hostpolicy and hostfxr DLLs cannot themselves be in the bundle (until statically linked singlefilehost is supported)._

Reply:
@wli3 I've added a new test `It_leaves_host_components_unbundled_when_necessary` for this. 

I thought about this, but I originally didn't add the test here because it is a temporary idiosyncrasy of the host/hostmodel, and not a requirement for the SDK. 
The host-tests do test for this:
https://github.com/dotnet/runtime/blob/master/src/installer/test/Microsoft.NET.HostModel.Tests/AppHost.Bundle.Tests/BundleExtractToSpecificPath.cs#L58
https://github.com/dotnet/runtime/blob/master/src/installer/test/Microsoft.NET.HostModel.Tests/Helpers/BundleHelper.cs#L63-L66

Reply:
If this is covered in runtime we can skip that. These are very slow tests * 6.

Maybe just have one test to ensure  

`ExcludedFiles = FilesToBundle.Zip(fileSpec, (item, spec) => (spec.Excluded) ? item : null).Where(x => x != null).ToArray();
` works. As in assert one file in one tfm that should be exclude not being there.

Reply:
And is all these TFM combination necessary? Again, these are really slow tests. There is no SDK logic have if-else about TFM. Only one test in SDK is enough. Other coverage should be in dotnet/Runtime

Reply:
OK @wli3. I'll keep the one interesting case of .net5 self-contained in the `It_rewrites_the_apphost_for_non_single_file_publish` test.

I think it is useful to have all the cases in the `It_runs_single_file_apps` case.

Reply:
@wli3 This is done in https://github.com/dotnet/sdk/pull/11586/commits/85c42def2678ee194d3c888ee5ca6d1e02d64189